### ISSUE TITLE
Code Completion not optimized behaviour: Code Completion gives the class constructor as  2nd, or 7th suggestion, instead of first.

### ISSUE THREAD

**Issue by CMMT20** [dart-lang/sdk#48600]
*I have already read the related threads, so i will summarize and reference what is already said in those.

Context: I'm building a flutter mobile app, using Android Studio, last version. When you are writing code, and have in mind the functionalities you want to build. Stopping to write, correct symbols and so stop your flow a lot, that is why this is relevant.

In the specific case of widgets, the Code Completion not desired behaviour is that gives more relevance to the widget name `Scaffold` than to the widget constructor `Scaffold()`. So the widget name appears first in Code Completion.
<img width="892" alt="1 autocompletion" src="https://user-images.githubusercontent.com/76145592/158856517-9ca464c8-d902-4764-bac1-422a31df8d86.png">

You can improve sorting by name the code completion, but still improvable:
<img width="901" alt="4 autocompletion sort by name" src="https://user-images.githubusercontent.com/76145592/158856760-39940b81-9eb4-48fb-a123-1a1081435719.png">

The key question is:
**_How can you get from Code Completion the class constructor as first suggestion, not as 2nd, or 7th_**

After asking to the IDE team
1 The order in these code complete suggestions is not given by the IDE software (Android Studio), but it is given by the dart sdk software
Reference:
https://intellij-support.jetbrains.com/hc/en-us/community/posts/4682230867346-Android-Studio-how-to-optimize-Code-Completion-behaviour-How-can-you-get-from-Code-Completion-the-class-constructor-as-first-suggestion-not-as-2nd

"Unfortunately it's not something that is controlled by the Dart plugin. Code completion, highlighting, quick fixes and other code insight features are powered by the Dart Analysis Server, a tool from the Dart SDK. IDE only shows what server asks it to show. For the reference, issues specific to the Dart Analysis Server are tracked in https://github.com/dart-lang/sdk/issues. Please feel free to submit a ticket there"

After reading the related threads:
2 The order is given by the relevance, that comes from the relevance tables.
These tables was made from code repositories samples some years ago.

So the key question goes to:
1 If the dart team sees this correct, class constructor is more relevant than class name 
to change the tables
2 If does not
how can you change, customize those tables in order to
- change the relevance to fit your code usage
- change the order in code completion suggestions

**Comment by scheglov** [comment]
@bwilkerson

**Comment by FMorschel** [comment]
I'm testing this on the latest stable, and I can't reproduce this on VS Code:

<img width="869" height="281" alt="Image" src="https://github.com/user-attachments/assets/4192f606-c930-4128-b0f5-2dfd01bea3f9" />

Or IntelliJ (see the `Scaffold` type is further down the list):

<img width="1000" height="536" alt="Image" src="https://github.com/user-attachments/assets/1b5f5183-f249-4224-9fc8-d13924a6d409" />

I think we can close this now. Or did I miss anything?

CC @bwilkerson

**Comment by bwilkerson** [comment]
Yes, I think we can close this with the assumption that we have improved the ranking.
